### PR TITLE
added new flags and vcf_profile tool option

### PR DESCRIPTION
**Pull request by zipho**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/tb-profiler/tb_profiler_vcf_profile.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<tool id="tb_profiler_vcf_profile" name="TB-Profiler VCF-Profile" version="2.4">
+    <description>Infer strain types and drug resistance markers from sequences</description>
+    <requirements>
+        <requirement type="package" version="2.4">tb-profiler</requirement>
```
Comment (bgruening) [inline tools/tb-profiler/tb_profiler_vcf_profile.xml]: @zipho have you seen that version 2.5 is out? https://anaconda.org/bioconda/tb-profiler

Comment (zipho) [reply tools/tb-profiler/tb_profiler_vcf_profile.xml]: @bgruening I am unaware of that. I will check the changelog..and perhaps a new PR or an update to this one there after.

Comment (pvanheus) [reply tools/tb-profiler/tb_profiler_vcf_profile.xml]: Hi @bgruening, I see @zipho updated to 2.5 (which involves changes in the code but no changes in the UI), so I think that addresses your review comment?

Comment (bgruening) [reply tools/tb-profiler/tb_profiler_vcf_profile.xml]: Thats cool and addresses it. We need to change the shed.yml to a suite I think as there are now two tools without breaking tool linage. I need to find the syntax for this and need time. If you can change the shed.yml that would be awesome. Thats the showstopper on my site, sorry :(

Comment (pvanheus) [reply tools/tb-profiler/tb_profiler_vcf_profile.xml]: I will look into the changes to the `.shed.yml` and see if I can come up with a suitable example.

Comment (pvanheus) [reply tools/tb-profiler/tb_profiler_vcf_profile.xml]: I have updated the `.shed.yml` to provide a "suite" of tb-profiler tools. I hope this is correct.

File: tools/tb-profiler/.shed.yml [hunk 2]
```diff
@@ -13,7 +13,32 @@ long_description: |
...
 remote_repository_url: https://github.com/galaxyproject/tools-iuc/blob/master/tools/tb-profiler
 type: unrestricted
+categories:
+  - Sequence Analysis
+
+repositories:
+  tb_profiler_profile:
+    description: "Predict M. tuberculosis lineage and drug resistance from sequences in FASTQ or BAM"
+    include:
+      - tb_profiler_profile.xml
```
Comment (bgruening) [inline tools/tb-profiler/.shed.yml]: Do you want to include the test file @pvanheus ?

File: tools/tb-profiler/tb_profiler_vcf_profile.xml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    </assert_contents>
+                </element>
+            </output_collection>
+        </test>
+    </tests>
+    <help><![CDATA[    
+Summary
+=======
+
+The pipeline aligns reads to the H37Rv reference using bowtie2, BWA or minimap2 and then calls variants using SAMtools. These variants are then compared to a drug-resistance database. We also predict the number of reads supporting drug resistance variants as an insight into hetero-resistance (not applicable for minION data).
```
Comment (bgruening) [inline tools/tb-profiler/tb_profiler_vcf_profile.xml]: Mh, this description does not fit to the tool isn't it? The tool consumes a vcf.gz file?

File: tools/tb-profiler/tb_profiler_vcf_profile.xml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        
+        #if $output_format == "csv"
+            --csv
+        #else if $output_format == "txt"
+            --txt
+        #end if
+
+    ]]>    </command>
+    <inputs>
+        <param name="vcf_input" type="data" format="vcf_bgzip" label="VCF"/>
```
Comment (bgruening) [inline tools/tb-profiler/tb_profiler_vcf_profile.xml]: Can this tool also consume vcf or just vcf.gz? How do I get this file?

File: tools/tb-profiler/.shed.yml (hunk 2 above)
Comment (pvanheus) [reply tools/tb-profiler/.shed.yml]: Ok, done.

File: tools/tb-profiler/tb_profiler_vcf_profile.xml (hunk 3 above)
Comment (pvanheus) [reply tools/tb-profiler/tb_profiler_vcf_profile.xml]: Cleanup up descriptions.

File: tools/tb-profiler/tb_profiler_vcf_profile.xml (hunk 4 above)
Comment (pvanheus) [reply tools/tb-profiler/tb_profiler_vcf_profile.xml]: I have adapted the tool to add plain VCF support.

**Comment by pvanheus** [comment]
@zipho I removed the `update_tbdb` step in the `tb_profiler_vcf_profile` tool because I believe that this confuses data provenance, as the [effect](https://github.com/jodyphelan/TBProfiler/blob/master/tb-profiler#L124) of `update_tbdb` could change depending on when it is run.

The updating of the TBDB is a job for a new tool and probably new data manager.

**Comment by pvanheus** [comment]
@bgruening please review the state of `.shed.yml`. If I understand correctly, the existing name needs to remain to preserve continuity. I added two repositories for the two distinct tools. This is duplication to some extent though - should `tb_profiler_profile` remain under the name `tbprofiler` or should it be added as a repository in the way that I have currently done?

**Comment by bgruening** [comment]
In my defense it was failing before :)

**Comment by bgruening** [comment]
And a new release is out ... they release fast!

https://github.com/bioconda/bioconda-recipes/pull/16849 ... We do not need to update to the latest version. I just want to point it out.

File: tools/tb-profiler/tb_profiler_profile.xml [hunk 5]
```diff
@@ -1,7 +1,10 @@
-<tool id="tb_profiler_profile" name="TB-Profiler Profile" version="2.4">
+<tool id="tb_profiler_profile" name="TB-Profiler Profile" version="2.5">
     <description>Infer strain types and drug resistance markers from sequences</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
     <requirements>
-        <requirement type="package" version="2.4">tb-profiler</requirement>
+        <requirement type="package" version="2.5">tb-profiler</requirement>
```
Comment (bgruening) [inline tools/tb-profiler/tb_profiler_profile.xml]: using macros here as well?

**Comment by pvanheus** [comment]
The attempt to add support for the `vcf_profile` command fell victim to this bug within the upstream tool: https://github.com/jodyphelan/TBProfiler/issues/52

To compensate for this, the support for `vcf_profile` has been backed up in the meantime.

**Comment by pvanheus** [comment]
@bgruening @zipho and others watching this: I have reduced the scope of this PR substantially, so it is now effectively only an update on the original `profile` command. When the `vcf_profile` command is fixed a new PR can be opened to support that.

Can you please check the `.shed.yml` and other items to ensure that things are done properly? Then we can propose this for merging.

**Comment by bgruening** [comment]
Sorry @pvanheus for all the pain you had with this. I will squash and merge the commits to reduce the size if that is ok.

Thanks @zipho and @pvanheus!

**Pull request merged**